solana/fuzz/src/fuzz_trimmed_amount.rs

```diff
@@ -0,0 +1,24 @@
+use honggfuzz::fuzz;
+use ntt_messages::trimmed_amount::TrimmedAmount;
+
+fn main() {
+    loop {
+        fuzz!(|data: &[u8]| {
+            // Manually discard data that doesn't fit our inputs.
+            if data.len() != 10 {
```

Comment:
this seems inefficient, as it randomly generates inputs and discards the ones that are too long or short. It would be faster (and more readable imo) if we wrote it like
```rust
fuzz!(|data: (u64, u8, u8)| { .. }
```

I think this should be supported based on https://docs.rs/honggfuzz/latest/honggfuzz/macro.fuzz.html (as it supports anything that implements https://docs.rs/arbitrary/latest/arbitrary/trait.Arbitrary.html)

Reply:
Makes sense, I'll look into it

Reply:
just curious - why not use this https://docs.rs/quickcheck/latest/quickcheck/ in conjunction with Arbitrary?

Reply:
honggfuzz seems to be more actively maintained (and built by google). Any reason to use quickcheck @RahulMaganti47 ?